internal/netconf/interfaces.go

```diff
@@ -127,20 +129,21 @@ func getEVPNIfaces(kb KnowledgeBase) []EVPNIface {
 
 	vrfTableOffset := 1000
 	for i, n := range kb.Networks {
-		if n.Underlay {
+		if n.Underlay != nil && *n.Underlay {
 			continue
 		}
 
+		vrf := int(*n.Vrf)
 		e := EVPNIface{}
 		e.Comment = versionHeader(kb.Machineuuid)
-		e.SVI.Comment = fmt.Sprintf("# svi (networkid: %s)", n.Networkid)
-		e.SVI.VLANID = n.Vlan
+		e.SVI.Comment = fmt.Sprintf("# svi (networkid: %s)", *n.Networkid)
+		e.SVI.VLANID = VLANOffset + i
```

Comment:
This seems awkward to me, @mwindower im sure you now better

Reply:
this change was needed because the metal-networker had its own metal network struct defined with an additional `Vlan`
but this was precalculated in the same way see https://github.com/metal-stack/metal-networker/pull/32/files#diff-99be339fef685166ed1fe90c82953da725ee6e7475ba682ab63b82c0c8528291L87